packages/backend-common/src/util/KubernetesContainerRunner.test.ts

```diff
@@ -187,6 +187,25 @@ describeIfKubernetes('KubernetesContainerRunner', () => {
     );
   });
 
+  it('should not close the original log stream', async () => {
+    const options: KubernetesContainerRunnerOptions = {
+      kubeConfig,
+      name,
+      namespace: 'default',
+    };
+    const containerRunner = new KubernetesContainerRunner(options);
+    const logStream = new PassThrough();
+    const runOptions: RunContainerOptions = {
+      imageName: 'alpine',
+      args: ['echo', 'hello world'],
+      logStream,
+    };
+
+    await containerRunner.runContainer(runOptions);
+
+    expect(logStream.closed).toBe(false);
```

Comment:
Strange, I thought I verified on my local machine that the test would be red. I'll check it out when I get home.

Reply:
Thanks for the watchful eye @vinzscam, messy on my part 👀.

Apparently the `.closed` that I previously used in my assertion, only returns `true` once all the data from the stream has been flushed. That didn't happen before the assertion was performed, so there was a race condition.

Now I'm checking for `.writableEnded` which returns `true` as soon as `.end()` is called, and I also dropped in an assertion for `.destroyed` in case the underlying API calls `.destroy()` for whatever reason. That should cover most bases when the stream would be closed.

Unfortunately I spotted another potential race condition that affects the test [here](https://github.com/backstage/backstage/blob/925cf5dd7dedfd734ae1d8e0a20931ef6de6e9a4/packages/backend-common/src/util/KubernetesContainerRunner.ts#L299). The `Promise.race` might resolve before the `logPromise` is resolved, in which case the test would incorrectly pass.

The only way I can think of to get around this is by applying a timeout to the test and if the logStream is closed before that timeout, that the test would then fail. Which is still quite dirty. With the current test implementation, the test at least fails on my machine when applying unchanged `KubernetesContainerRunner.ts`.